### PR TITLE
fix: blend node and blink code generation policy when both are loaded

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change

This fixes a crash in some situations when sandbox: false.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed a crash that could occur when running eval in inline scripts in unsandboxed renderer processes.

### CODE REVIEW

File: shell/common/node_bindings.cc [hunk 1]
```diff
@@ -158,19 +158,37 @@ void V8FatalErrorCallback(const char* location, const char* message) {
...
       return false;
     return blink::V8Initializer::WasmCodeGenerationCheckCallbackInMainThread(
-        context, v8::String::Empty(isolate));
+        context, source);
   }
 
-  return node::AllowWasmCodeGenerationCallback(context,
-                                               v8::String::Empty(isolate));
+  return node::AllowWasmCodeGenerationCallback(context, source);
+}
```
Comment (MarshallOfSound) [inline shell/common/node_bindings.cc:173]: This change seems unrelated, other than being technically correct was there a reason for this change being tied to the current one?

File: shell/common/node_bindings.cc [hunk 2]
```diff
@@ -158,19 +158,37 @@ void V8FatalErrorCallback(const char* location, const char* message) {
...
+
+v8::ModifyCodeGenerationFromStringsResult ModifyCodeGenerationFromStrings(
+    v8::Local<v8::Context> context,
+    v8::Local<v8::Value> source,
+    bool is_code_like) {
+  // If we're running with contextIsolation enabled in the renderer process,
```
Comment (MarshallOfSound) [inline shell/common/node_bindings.cc:179]: Not sure which part of this logic is specific to ctx isolation, looks like we ask blink in all non-browser processes?

Comment (MarshallOfSound) [reply shell/common/node_bindings.cc:179]: Does this work in the new utility processes?

File: shell/common/node_bindings.cc (hunk 1 above)
Comment (nornagon) [reply shell/common/node_bindings.cc:173]: No, this is incidental.

File: shell/common/node_bindings.cc (hunk 2 above)
Comment (nornagon) [reply shell/common/node_bindings.cc:179]: when context isolation is disabled, there is a node env for the main world, so we get the node behavior for this function.

there's no blink in the utility process and there is a node env for the context there so this works just the same as main there.

**Review by MarshallOfSound — APPROVED** [review]

File: shell/common/node_bindings.cc [hunk 3]
```diff
@@ -158,19 +158,35 @@ void V8FatalErrorCallback(const char* location, const char* message) {
...
+  // fall back to Blink's logic.
+  if (node::Environment::GetCurrent(context) == nullptr) {
+    if (gin_helper::Locker::IsBrowserProcess()) {
```
Comment (deepak1556) [inline shell/common/node_bindings.cc:181]: Which scenario we support today might enter this block ? Don't we always have a node environment in the browser process ?

**Review by deepak1556 — APPROVED** [review]

File: shell/common/node_bindings.cc (hunk 3 above)
Comment (nornagon) [reply shell/common/node_bindings.cc:181]: None, hence the NOTREACHED below

Comment (deepak1556) [reply shell/common/node_bindings.cc:181]: Do we even need the `NOTREACHED` in that case ? My understanding is that `NOTREACHED` is useful to abort when there is no `default` action for switch statements or when an method is not implemented.

Comment (nornagon) [reply shell/common/node_bindings.cc:181]: I guess this could just be a CHECK :)
